immuni_analytics/apis/analytics.py

```diff
@@ -0,0 +1,276 @@
+    authorize_analytics_token,
+)
+from immuni_analytics.celery.authorization.tasks.verify_safety_net_attestation import (
+    verify_safety_net_attestation,
+)
+from immuni_analytics.celery.operational_info.tasks.store_operational_info import (
+    store_operational_info,
+)
+from immuni_analytics.core import config
+from immuni_analytics.core.managers import managers
+from immuni_analytics.helpers import safety_net
+from immuni_analytics.helpers.redis import get_authorized_tokens_redis_key_current_month
+from immuni_analytics.models.operational_info import OperationalInfo as OperationalInfoDocument
+from immuni_analytics.models.swagger import (
+    AppleOperationalInfo,
+    AuthorizationBody,
+    GoogleOperationalInfo,
+)
+from immuni_common.core.exceptions import SchemaValidationException
+from immuni_common.helpers.sanic import json_response, validate
+from immuni_common.helpers.swagger import doc_exception
+from immuni_common.models.enums import Location, Platform
+from immuni_common.models.marshmallow.fields import (
+    Base64String,
+    IntegerBoolField,
+    IsoDate,
+    Province,
+)
+from immuni_common.models.swagger import HeaderImmuniContentTypeJson
+
+_LOGGER = logging.getLogger(__name__)
+
+bp = Blueprint("analytics", url_prefix="analytics")
+
+
+@bp.route("/apple/operational-info", methods=["POST"], version=1)
+@doc.summary("Upload operational info (caller: Mobile Client.)")
+@doc.description(
+    "Check if the analytics_token is authorized to upload and save the operational info"
+    " in the database."
+)
+@doc.consumes(doc.Boolean(name="Immuni-Dummy-Data", required=True), location="headers")
+@doc.consumes(AppleOperationalInfo, location="body")
+@doc.consumes(HeaderImmuniContentTypeJson(), location="header", required=True)
+@doc.consumes(
+    doc.String(name="Authorization", description="Bearer <ANALYTICS_TOKEN>"),
+    location="header",
+    required=True,
+)
+@doc_exception(SchemaValidationException)
+@doc.response(
+    HTTPStatus.NO_CONTENT.value, None, description="Well-formed request.",
+)
+@validate(
+    location=Location.HEADERS,
+    is_dummy=IntegerBoolField(
+        required=True,
+        data_key="Immuni-Dummy-Data",
+        allow_strings=True,
+        description="Whether the current request is a dummy request. Dummy requests are ignored.",
+    ),
+)
+@validate(
+    location=Location.JSON,
+    province=Province(),
+    exposure_permission=IntegerBoolField(required=True),
+    bluetooth_active=IntegerBoolField(required=True),
+    notification_permission=IntegerBoolField(required=True),
+    exposure_notification=IntegerBoolField(required=True),
+    last_risky_exposure_on=IsoDate(),
+)
+async def post_operational_info(  # pylint: disable=too-many-arguments
+    request: Request,
+    is_dummy: bool,
+    province: str,
+    exposure_permission: bool,
+    bluetooth_active: bool,
+    notification_permission: bool,
+    exposure_notification: bool,
+    last_risky_exposure_on: date,
```

Comment:
It might be worth to handle all of the operational infos as a single object `OperationalInfo` as in the endpoint logic we don't really need to handle any of this data alone (we simply forward it right?) 

This would allow to increase readability and decrease the amount of parameters of all functions.

The simplest implementation that comes into my mind would be `namedtuples`, or a dataclass with a special validator that expands the json response (similar to how we handle the ExposureDetectionSummary in ingestion

Reply:
Is it worth though? We put them in redis immediately after.